src/RestApi/Utilities/ProductSummary.php

```diff
@@ -0,0 +1,79 @@
+<?php
+/**
+ * Helper class to format a short summary of content for a product.
+ *
+ * @package WooCommerce/Blocks
+ */
+
+namespace Automattic\WooCommerce\Blocks\RestApi\Utilities;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Product Summary class.
+ */
+class ProductSummary {
+	/**
+	 * Original content from the product.
+	 *
+	 * @var string
+	 */
+	protected $content = '';
+
+	/**
+	 * Constructor.
+	 *
+	 * @param \WC_Product $product The product to generate a summary for.
+	 */
+	public function __construct( \WC_Product $product ) {
+		$this->content = $product->get_short_description();
+
+		if ( ! $this->content ) {
+			$this->content = $product->get_description();
+		}
+	}
+
+	/**
+	 * Return the formatted summary.
+	 *
+	 * @param int $max_words Word limit for summary.
+	 * @return string
+	 */
+	public function get_summary( $max_words = 25 ) {
+		$summary = $this->content;
+
+		if ( $max_words && $this->get_word_count( $summary ) > $max_words ) {
+			$summary = $this->generate_summary( $summary, $max_words );
+		}
```

Comment:
We could but it won't benefit us in this case due to only calling it once per product.

Reply:
Right, something to keep in mind if this class ever lands in Woo Core for more general use.

> Also we'd need to account for the $max_words you can pass in which could change using the same instance.

Yup, I was thinking the cache would be an array indexed by the value of `$max_words`.  But agree with not doing anything in this pull because for now, it's unnecessary.